File: 475. Heaters/Python/Solution.py
class Solution(object):
    def findRadius(self, houses, heaters):
        """
        :type houses: List[int]
        :type heaters: List[int]
        :rtype: int
        """
        houses = sorted(houses)
        heaters = sorted(heaters)
        heater_index = 0
        result = 0
        for house in houses:
            while heater_index + 1 < len(heaters) and abs(house - heaters[heater_index]) >= abs(
                    heaters[heater_index + 1] - house):
                heater_index += 1
            result = max(result, abs(house - heaters[heater_index]))
        return result

File: 475. Heaters/C++/Solution.cpp
class Solution {
public:
    int findRadius(vector<int> &houses, vector<int> &heaters) {
        sort(houses.begin(), houses.end());
        sort(heaters.begin(), heaters.end());
        int heaterIndex = 0;
        int result = 0;
        for (int house : houses) {
            while (heaterIndex + 1 < heaters.size() &&
                   abs(house - heaters[heaterIndex]) >= abs(heaters[heaterIndex + 1] - house)) {
                heaterIndex++;
            }
            result = max(result, abs(house - heaters[heaterIndex]));
        }
        return result;
    }
};